llvm/lib/Target/AArch64/AArch64ISelDAGToDAG.cpp

```diff
@@ -5098,6 +5099,12 @@ void AArch64DAGToDAGISel::Select(SDNode *Node) {
                AArch64::LUTI2_4ZTZI_S}))
         // Second Immediate must be <= 3:
         SelectMultiVectorLuti<3>(Node, 4, Opc);
+      else if (auto Opc = SelectOpcodeFromVT<SelectTypeKind::FP>(
```

Comment:
Can you instead just pass `<SelectTypeKind::AnyType>` in the if statement above?

Reply:
Yeah that does seem better. I'm not sure what the implications of letting i1, i64 and f64 are though.

Reply:
If AnyType is used and i1 is passed, `SelectOpcodeFromVT` will return 0.
I think it would be the same for i64/f64 as the size of `Opcodes` for these intrinsics is 3 and the function will check this when trying to return the correct opcode from the list:

`return (Opcodes.size() <= Offset) ? 0 : Opcodes[Offset];`